what would it take to pull in upstream changes???

rossgrady opened the pull request.

mw9 committed:
Slim::Formats: Fix up logging - missing log category

Slim::Formats sets up its log with:

    my $log  = logger('formats');

But the logger category 'formats' does not exist, so it does not show up
as an available logging option.

This change directs logging output to the pre-defined server category
'formats.audio', which is probably most appropriate.

An alternative would be to set up the pre-defined category 'formats'.
But the additional granularity afforded by this approach is unlikely to
be useful.

This issue appears to have been in existence since the adoption of
Log4perl debugging.

Refer commit 6f27c52f 'Implement Log::Log4perl debugging'.

mherger committed:
Don't corrupt multi-byte Unicode characters. This is an old bug in an
even older Perl module. Let's patch it.

https://rt.cpan.org/Public/Bug/Display.html?id=9700

mw9 committed:
InternetRadio plugin: Fix up logging - missing log category

The logger category ‘plugin.radio’ does not exist, so it does not show
up as an available logging option. We need to create the category.

mherger committed:
Update HTML::FormatText to fix compatibility with newer Perl versions.

mherger committed:
Running system_profiler is slow. Don't run it twice. This shaves off a few hundred milliseconds on every restart. Yeah!

mw9 committed:
Slim::Networking::Async: Improve error diagnostic

Add some diagnostic output if there is socket creation or connection
failure.

Makes it much easier to figure out the problem.

mw9 committed:
Slim::Utils::Log: Missing log category 'player.ui.screensaver'

Slim::Buttons::ScreenSaver sets up its log with:

    my $log  = logger('player.ui.screensaver');

But the log category 'player.ui.screensaver' does not exist, so it does
not show up as an available logging option.

This change sets up the category with a default log level of 'ERROR'.

This issue appears to have been in existence in some form since the
adoption of Log4perl debugging.

Refer commit 6f27c52f 'Implement Log::Log4perl debugging'.

mw9 committed:
Formats::XML - Podcasts: Fix iTunes image tag recognition

An itunes:image tag exists at the channel level of the feed, *not* the
top level.

We retain the existing (historic) check at the top level in case some
feeds still use that approach. But we ensure that we have the expected
hash ref.

Refer:
https://help.apple.com/itc/podcasts_connect/#/itcb54353390

mw9 committed:
Formats::XML - Podcasts: Badly formed RSS image URL breaks Jive

A broken podcast may provide a malformed image url tag. If, for any
reason, we get an empty url tag, or worse, XMLin may interpret it as
a (possibly empty) hash reference, or something else other than a scalar
value. This will break podcast browsing in Jive.

So we add some checks to our retrieved image URLs to ensure that they
are, indeed, scalars. The appended (historic) example is illustrative.

An alternative solution might be available by adapting the parameters
passed to XMLin (in xmlToHash) that control its rendering of the XML.
But that may inadvertently break some other aspect of the parsing.

Example:

Observed in September 2017, although no longer present in the errant
feed. The (edited) feed was as follows:

    <channel>
        <title>A title</title>
        <description>Some description</description>
        <link>A_Link.com</link>
        <pubDate>Sometime in Sept 2017</pubDate>
        <image>
            <url> </url>
            <title>Another title</title>
            <link>Another_Link.com</link>
        </image>
    etc.

If the channel/image/url tag is simply empty, as above, xmlToHash will
return $xml->{'channel'}->{'image'}->{'url'} as an *empty hash ref*,
( image->url = {} ) and *not* a zero length string, or 'undef', as one
might expect.

Jive browsing then breaks when attempting to fetch artwork in
SlimBrowserApplet, because it requires that the icon variable be a
string. It breaks on the structured icon object, even though empty. The
podcast browse menu is not properly rendered, and presents a bewildering
display. Not a good user experience.

mw9 committed:
Slim::Schema::RemoteTrack: Tidy up debug logging of undefined attribute

This change tidies up a debug log message.

When $self->$attribute() is undefined we get warning noises:
    Use of uninitialized value in join or string at .... etc.

It's more helpful to have an explicit statement that the attribute being
gotten is simply undefined.

warthog9 committed:
Fix suprious error message relating to SSL_verify_mode

So SSL_VERIFY_NONE was the old default in IO::Socket::SSL however
at one point they started warning that this was a rather insecure
thing to do.  The workaround was to explicitly set
SSL_verify_mode => SSL_VERIFY_NONE from the same package.  This
posed some oddities, both because SSL_VERIFY_NONE as a bare word
wasn't exported well into slimserver, and because it's not
entirely obvious where that's coming from.  It was eventually
noticed that SSL_VERIFY_NONE didn't always work and 0x00 was
hard coded, EXCEPT this doesn't work on some other systems
where SSL_VERIFY_NONE is not 0x00.

Since the check in question comes out of IO::Socket::SSL and it
follows:
	! $is_server
        and ! $arg_hash->{SSL_reuse_ctx}
        and ! exists $arg_hash->{SSL_verify_mode}
        and $default_args{SSL_verify_mode} == SSL_VERIFY_NONE

where SSL_VERIFY_NONE is defined as:
	use constant SSL_VERIFY_NONE => Net::SSLeay::VERIFY_NONE();

This patch just accepts that Net::SSLeay::VERIFY_NONE() is what we should
have been setting this too all along.

As stated in a previous commit SSL_VERIFY_NONE

Signed-off-by: John 'Warthog9' Hawley <warthog9@eaglescrag.net>

mherger committed:
Fix condition to give streaming some slack: only idle every x iterations, not all but the xth iteration.

mherger committed:
Merge pull request #222 from ptrunley/keepalive

Fix logic for non-keepalive connections

mherger committed:
Merge pull request #203 from philippe44/syncVolume

only use synchronized volume players

mherger committed:
Add support for Perl 5.28 on Linux x86_64

mherger committed:
Fix issue with duplicated variable definition no longer accepted by Perl 5.28

mherger committed:
Fix sort order of index list. We need to utf8 decode the value, or its encoding will lead to invalid display.

mherger committed:
Add a way to preserve the shuffled playlist order. This comes in handy with features like Don't Stop The Music (where we want to add tracks without re-shuffling), or replacing local tracks with their volatile URL before a full scan is run. Greatly simplifies and improves the latter use case.

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Fix "Updated Software" page handler to report plugin names

mherger committed:
Add JSON version of software update page handler.

mherger committed:
Updated Software page would sometimes return the full list of installed plugins, rather than just the updated list. Share some code with the Settings pages which handled updates correctly already.

mherger committed:
Add option to exclude paths, rather than include them.

mherger committed:
Add support for text match exclusion in Advanced Search

mherger committed:
Add "starts/doesn't start with" options to text searches in advanced search

philippe44 committed:
enable "All Songs" for simple OPML based plugins

mherger committed:
Merge pull request #225 from philippe44/public/7.9

enable "All Songs" in Deezer and simple OPMLBased plugins

philippe44 committed:
use playall attribute in 'All Songs" only when set to true

ralph-irving committed:
Cygwin sox.exe fails to traverse windows file paths.

mw9 committed:
SimpleAsyncHTTP: Don’t cache stale documents - they are never revalidated

Some (older ?) servers may signal a ‘don’t cache’ request by simply returning
the header ‘expires: Thu, 01 Jan 1970 00:00:00 GMT’, without returning any
‘Cache-Control’ headers. Such a document will appear to be permanently stale.
Unwise and very out of date behaviour, perhaps, but it can still be seen in the
wild.

In these circumstances, SimpleAsyncHTTP will seek to cache the document for a
negative period of time ($expires < 0). Unfortunately the effect of this is to
permanently cache the document without ever subjecting it to revalidation.

So, for example, updates to a podcast feed might never be seen unless/until LMS’
cache is cleared. This is not desired behaviour.

Reasons:

 a) The document will never be revalidated, because it passes the
    ‘no revalidaton’ optimization ($expires < $max).
 b) Slim::Utils::Cache/DbCache treats a negative caching period as meaning
    “Never expire”.

The combined effect is that such a document will live permanently in the cache,
and the latest version of the document will never be revalidated/fetched.

This behaviour is not limited to documents served with the header
‘expires: Thu, 01 Jan 1970 00:00:00 GMT’. Any document that appears to be stale
will be so treated. A misconfigured server could produce such a result.

This proposed change deals with the issue by detecting such ‘apparently stale’
documents, and simply not caching them.

Remark: The caller may set an explicit expiration time ($params->{expires}). A
negative value will be ignored.

paul-1 committed:
Update Perl 5.24 armhf to Audio::Scan 1.02

mherger committed:
Fix web UI playlist with remote streams.

Don't cache single track playlists, as they would update metadata on the fly. And the penalty for re-rendering a page with a single track isn't too hard to take.

mherger committed:
Log error when DSTM decides not to start because SugarCube or RandomPlay was active.

mherger committed:
Enable https streaming for Slacker. While the regular audio files are http only, the ads are https...

mherger committed:
Revert previous attempt at fixing the playback issue on Slacker.

Commit:
CHG: Some Podcastes (like "Lage der Nation") offers multiple images ...

Commit:
CHG: Some Podcastes (like "Lage der Nation") offers multiple images ...

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Tone down warning about DSTM interfering with RandomPlay/Sugarcube

mherger committed:
Don't overwrite RandomPlay's genre selection when using RandomPlay as the Don't Stop The Music mixer

mherger committed:
Add the dontContinue parameter to randomplay:// URIs. They allow to queue up a limited number of random items without automatically adding more.

Update outdated documentation about the RandomPlay CLI commands.

mherger committed:
Add simple Presets-editor to configure the Squeezebox' Preset buttons

mherger committed:
Add copyright notice to re-factored RandomPlay modules.

mherger committed:
Don't fail Presets page rendering when there's no valid client.

Tweak layout to be more table like.

mherger committed:
Fix typo

mherger committed:
Merge branch 'public/feature/PresetsEditor' into public/7.9

jjvdgeer committed:
Expand range for alarm timout to 240 minutes

mherger committed:
When staring a RandomPlay mix from a Genre, don't change the genre selection for other mixes.

mw9 committed:
Formats::XML - Podcasts: Use episode specific image

At present, each episode of a podcast is displayed in the various browse
menus with an accompanying image. This image is taken from the 'channel'
level of the underlying RSS feed, and is identical for each episode. The
standard RSS format makes no provision for images to be provided at the
'item' (episode) level.

But Apple/iTunes extends the format to, optionally, provide a separate
image for each individual episode of a podcast. LMS could make use of
these images, thereby adding a little more variety to the feed display.

This proposed change achieves that.

The episode image, if it exists, is found in the 'href' attribute of an
'itunes:image' tag, at the 'item' level of the feed.

Reference:
https://help.apple.com/itc/podcasts_connect/#/itcb54353390

mherger committed:
Search for a term including an asterisk ("*example*") could lead to all tracks/albums being returned. Which eventually could bring a lower powered LMS system to a halt.

mherger committed:
Change presets on text change only, too.

mw9 committed:
Formats::XML - Podcasts: Minor clarifications to code

Conventional RSS feeds provide a single ‘image’ element containing,
amongst other things, a URL which points to the image. ‘XMLin’ will
provide us with a hash ref in this circumstance.

Some RSS feeds provide multiple image elements. In this case ‘XMLin’
will provide an array of such ‘image’ elements. We arbitrarily pick
the first one.

Refer original commit: baea48ee650bd63effeab97bfd87f94801185388

mherger committed:
Fix accidental removal of an import.

mherger committed:
Remove preset if URL has been removed

philippe44 committed:
DSF vs DFF in sample_size

mherger committed:
Update changelog.

alexte committed:
Blocked CLI connect killed CLI listening port

on "CLI is restricted" the code closed the listening port instead of the accepted port.

mherger committed:
Make sure we correctly URI encode/decode the genre names.

mherger committed:
Try to fix path encoding in Media Dirs web UI

mherger committed:
Don't store apps settings for players which have never been connected to this server instances.

mherger committed:
Fix typo in CLI documentation.

mherger committed:
Don't create new pref entries for players this server has never seen when processing mysb.com apps.

mherger committed:
Use operating system identification file /etc/os-release to identify Linux flavour.

mherger committed:
Leave a note about how to make sure we do include the migration files with the Windows build.

mherger committed:
Fixes #237 - do not cross-fade single tracks in repeat mode

philippe44 committed:
fade correction

mherger committed:
Add README.md

philippe44 committed:
initial commit

mherger committed:
Update PAR to be compatible with newer Perl versions

We rarely use it any more, but there still are some plugins using it.

mherger committed:
Add German translation for newly added Fade-In option

mherger committed:
Update changelog

puck committed:
When there are plugins to be updated, use "Update" not "Enable" for column header

One of those little niggles that I always notice is wrong when I have to
update plugins!

mherger committed:
Add some more localizations for Update

mherger committed:
Display plugin update information, even if it's only a single string, not a list of localizations.

maniac103 committed:
Always treat radio streams as playlists.

When not doing so, on startup the radio stream M3U will be attempted to
be played, which (obviously) fails.

maniac103 committed:
Preserve playing state when client vanishes.

Store 'playing at power off' pref if a playing client vanishes (e.g.
because it's powered down) so we can restore playing state when it
reappears.

maniac103 committed:
Run client startup before powering it up.

Doing so is important for restoring the client playlist before powering
up, which will use the playlist.

maniac103 committed:
Make sure Player::power is run from Player::init.

Previously the power() call was mostly a no-op, as $currOn always equals
$on when called from Power::init, as both values are loaded from the
same preference.

mw9 committed:
strings.txt: Descriptions for player.ui.screensaver logging

This change adds descriptive texts for the newly created
'player.ui.screensaver' log category.

These were omitted from the relevant commit 3c6b8599:
'Slim::Utils::Log: Missing log category player.ui.screensaver'.

mherger committed:
Minor tweak to mania103's playlist discovery mode.

mherger committed:
Add more log categories to the scanner logging group so that the scanner does actually pick them up.

Thanks mw9 for the patch!

Fixes #250 #253

mherger committed:
Add missing comma...

mherger committed:
Use $track->{secs} as alternative to $trac->{duration} as some services seem to ignore the latter.

ItaintYellow committed:
Dutch translations -  update

adding missing translations, minor linguistic corrections and typo's. Terminology changes.

Fixes #254

philippe44 committed:
add socks proxy

grammar!

bad process

cleaning

cleaning

cleaning

adding SSL through SOCKS as well

cleaning

socks4 by default, error management, put HTTPSSocks at the right place

add SOCKS5 authentication

better reuse IO::Socket::Socks + add obf/unobf

remove obfuscate function

set HTTP 1.0

philippe44 committed:
enable HTTP 1.1

mherger committed:
Changelog update

mherger committed:
Fix/enable using the the image proxy with the resize daemon

mherger committed:
Allow passing the image data do the resizing daemon. No need to write it out to a file and read it in again.

paul-1 committed:
Add support for Perl 5.28 on Linux armhf

mherger committed:
Fix resizing using the daemon: base64 encode the data before the transfer, don't send the length, as it could interfere with the connections expectation to hung up on cr/lf.

mherger committed:
Add trackinfo items for play count and last played information.

mherger committed:
Add more Synology specific folders to filter out

mherger committed:
More Synology specific folder filtering

mherger committed:
When scanning a single folder, don't scan all artwork, but only artwork for the selected folder.

mherger committed:
More Synology folders to hide.

mherger committed:
Fix #261 Use non-blocking Net::HTTPS::NB instead of Net::HTTP

mherger committed:
Don't crash the RandomPlay DSTM integration on non-track objects.

Try to get a track's object in case we only receive its URL.

mherger committed:
Changelog update

mherger committed:
Changelog update

mherger committed:
More tweaks to the Synology support: show DSM version, set default for music folder, remove support for legacy systems (SSOTS, NSLU2)

mherger committed:
Sometimes the "touch to play" menu would get confused if one of the items was a code reference. This hack is working around this.

mavit committed:
Add a simple EditorConfig file

Tell text editors that our Perl files use tabs for indentation.  See https://editorconfig.org/ for more details.

mavit committed:
Non-blocking TLS handshake

mherger committed:
Merge pull request #264 from mavit/editorconfig

Add a simple EditorConfig file

mherger committed:
Merge pull request #265 from mavit/non-blocking-https

Non-blocking TLS handshake

mherger committed:
Try to guess the image file type from the magic bytes in case the web server is returning text/plain.

mherger committed:
Add Perl 5.30 support for Linux x86_64

mherger committed:
When bringing up the track info page in the web UI, try to get a radio streams station URL rather than the streaming URL, as the latter can change.

mherger committed:
Add Perl 5.28 binaries for aarch64-linux, remove some cruft from recently added 5.30 build

Commit:
Change to detect MPEG-4 files with leading mdat. These files have the audio index at the
end of the file and so cannot be played streaming. Player with native AAC/ALAC such as
Radio, Touch, squeezelite when trying to play these file will log LMS error
"Decoder does not support file format, code 0".
This change creates new audio types alcx and mp4x for these files and so will always
be transcoded. The new types appear in FileTypes settings as "MP-4 leading mdat"
"Apple Lossless leading mdat"

Users can "fix" these files by using ffmpeg "movflags +faststart" option.

Commit:
Changes File type text to be more easily understood/less technical

mherger committed:
Add More and Insert buttons to Advanced Search results.

mherger committed:
Merge pull request #270 from bpa-code/public/7.9

Change to detect MPEG-4 files with leading mdat. These files have the…

mherger committed:
Changelog update

mherger committed:
Move menu position cache to client object, and get rid of some obsolete code.

mherger committed:
Get rid of some unused code.

mherger committed:
Remove SqueezeOS related code. This code base will never be integrated with some SB firmware.

mherger committed:
Skip TuneIn stations which are labeled 'unavailable'

philippe44 committed:
add OPUS

mherger committed:
Synology's DSM is creating a never ending list of folders to ignore. Try to catch them with a simple regex...

mherger committed:
Fix #275 - fix encoding in `readdirectory` query

mherger committed:
Wipe BMF cache when needed

mherger committed:
Fix readdirectory query for Windows, too...

mherger committed:
Add a log statement when we're done resizing an image.

mherger committed:
Fix some issues with the templating system

* increase WHILE_MAX from 1000 to 10000 - some larger collections would fail to render the Advanced Search page with lots of genres
* increase MAX_DIRS to 256
* wipe disk cache on LMS updates to prevent failure due to outdated templates

mherger committed:
Changelog update

mherger committed:
Fix #276 - don't try to read cache folder before it's been created

mherger committed:
Remove test code

mherger committed:
Sometimes readdirectory wouldn't list folders when trying to access a drive's root, by only requesting data from eg. c: rather than c:\. For whatever reason Perl or Windows or whatever at some point started to return content of the current directory on given drive, rather than just the root. Let's work around this by adding a slash in case only the drive letter is given.

mw9 committed:
Enable OggFlac playback from remote playlist

At present, LMS supports playback of remote OggFlac streams directly.
But it does not support playback of an OggFlac stream listed in a remote
playlist. Playback silently fails. This has been observed with both
'PLS' and 'M3U' remote playlist formats.

This change remedies matters by adding an 'ogf' content type id to LMS's
global 'types.conf' file. Only the 'ID' and 'Server File Type' fields
are relevant, 'Suffix' and 'Mime Content-Type' fields are left inactive.

OggFlac streaming support was added to LMS in 2012. One feature of the
change is that a new audio type id, 'ogf', was created to distinguish
the flac variant, which uses a different decode path.

The original change may be found here:
bug 12893: Support OggFLAC - server support
https://github.com/Logitech/slimserver/commit/a8464b42

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Do not scan #snapshot folders on DSM - there might be dozens of them...

mherger committed:
Truncate scanned_files table from library.db at optimization time. It can take up to 25% of the file's size, but isn't needed once the scan is done.

mherger committed:
Fix #278 - correctly encode password before sending it to mysb.com for authentication

mherger committed:
Fix #278 - correctly encode password before sending it to mysb.com for authentication

mherger committed:
Merge branch 'public/7.7' into public/7.9

# Conflicts:
#	Changelog7.html

mherger committed:
Add support for macOS 64-bit binaries

mherger committed:
Unify handling of search path for helper applications

Binaries inside plugin sub-folders followed a different logic than those for the globally available binaries. Eg. "darwin" wouldn't be searched for plugins, but globally only.

mherger committed:
Use $baseDir rather than default, hard-coded folder

mherger committed:
Improve architecture detection on macOS. Under certain circumstances we'd not get it from system_profiler.

mherger committed:
Fix bad merge/cherry-pick

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Remove plugins for defunct services (Live365, Flickr)

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Remove code references to Live365

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Patch AnyEvent::Socket. It's causing errors on newer Perl versions.

mherger committed:
Add another patch to AnyEvent::Handle. It's causing errors on newer Perl versions.

mherger committed:
Future versions of macOS will come without Perl pre-installed. The binary might live elsewhere.

Add some later additions to the cleanup too.

mherger committed:
Add the server's IP address to the serverstatus query response.

mherger committed:
Add the LMS web port to the serverstatus query response.

mherger committed:
Update doc for serverstatus query

Commit:
Initial attempt to sort 'All Songs' using album years (if set to do so)

Commit:
Place 'sort' into '$what'

Commit:
Update CLI docs adding sort parameter to playlistcontrol

mherger committed:
Add faad binary for aarch64

Commit:
Remove sortParam - no longer used.

Commit:
Remove TODO

Commit:
Only set sort to albumYearSort if year biased album sort passed in and
sort was albumSort

mherger committed:
Ignore sort type "albumtrack", too.

mherger committed:
Merge pull request #290 from mw9/feature/ogg_flac_in_playlist

OggFlac playback from remote playlist silently fails

mherger committed:
Fix #294 - status query would return incomplete tracklist if a track did not exist in the persistent table

Use a left join to include all tracks in the query.

mherger committed:
Don't lock up LMS while purging caches.

It can lead to timeouts on startup interaction with mysb.com and otherwise delay the startup procedure. I've seen lockups of 15-20 seconds while purging a largish image proxy cache file.

mherger committed:
Add logging category for the update checker.

philippe44 committed:
HTTP 1.1 enablement

philippe44 committed:
only add http_version

philippe44 committed:
Check version

mherger committed:
Merge pull request #295 from philippe44/public/7.9

HTTP 1.1 enablement

philippe44 committed:
HTTP 1.1 continuation

paul-1 committed:
perl 5.28 - armhf: Rebuild Image::Scale and Media::Scan with libpng 1.6.37

mherger committed:
Merge pull request #297 from piCorePlayer/public/7.9

perl 5.28 - armhf: Rebuild Image::Scale and Media::Scan with libpng 1…

mherger committed:
Merge pull request #296 from philippe44/public/7.9

HTTP 1.1 continuation

mherger committed:
Fixes #298 - lower the log level on disconnected clients warning

philippe44 committed:
Suspend/Resume HTTP streaming

mherger committed:
Merge pull request #299 from philippe44/public/7.9

Suspend/Resume option for HTTP streaming

mherger committed:
Keep the style: don't mix camel case with snake case.

mherger committed:
Revert commit f7c9692 - it breaks AutoRescan

karcherm committed:
Fix logic for artist display in track list

The comparison whether the track artist is different from the album artist must not compare escaped to unescaped strings.

mherger committed:
Merge pull request #300 from karcherm/patch-1

Fix logic for artist display in track list

mherger committed:
Improved OGF stream information - better estimate the stream's bitrate (in particular for hires streams).

mherger committed:
Update copyright notices

mherger committed:
Fix target application for a few plugins

mherger committed:
Bump changelog

mherger committed:
LMS 7.9.3!

mherger committed:
Improve macOS detection: newer versions are 64 bit, even if the architecture string is wrong or missing (eg. when run in a VM).

mherger committed:
Fix JS in the Presets editor: the function must be assigned to a global variable to be accessible from a dynamically loaded HTML page.

See eg. https://stackoverflow.com/questions/1812560/prototype-ajax-updater-eval-javascript-functions

mherger committed:
Fix release date in Changelog

mherger committed:
Fix a crasher on Windows when TLS handshaking would time out

mherger committed:
Improve Raspbian discovery. Need to check in /etc/os-version, too. Newer versions seem to no longer come with /etc/debian_version etc.

mherger committed:
Fix typo in linux distribution detection

mherger committed:
Fix #307 - add more MIME type definitions for WAV files

mherger committed:
Remove redundant MIME type audio/x-wav

mherger committed:
Strip BOM from first line in .m3u files, no matter whether it's a comment or not.

mherger committed:
Changelog update

mherger committed:
When using the CT tag in the player UI to display the content type, use the same information as we're using in the `status` query. Fall back to `type` for remote streams.

mherger committed:
Make sure we update all content type caches

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Back-port Slim::Schema->clearContentTypeCache from 7.8+

mherger committed:
Merge branch 'public/7.7' into public/7.9

# Conflicts:
#	Slim/Schema.pm

philippe44 committed:
OggOpus detetcion and content-type adjustment

philippe44 committed:
Allow fade-in for squeezeplay client that don't handle it locally when < 1s

mherger committed:
Update changelog

mherger committed:
When a plugin fails to install, get more information out of it

* log expected and actual checksum
* if in debug mode, copy downloaded file to cache folder for further investigation

ralph-irving committed:
Skip version/revision check for jivelite as all releases support the Defeat destructive Touch-To-Play feature.

mherger committed:
Prepare for TIDAL's upcoming move to AAC

mherger committed:
Try to parse bitrate from TIDAL's stream headers

mherger committed:
Don't hard-code the Pandora bitrate - it could be wrong.

mherger committed:
Merge branch 'public/7.7' into public/7.9

mherger committed:
Remove some cruft from the Pandora ProtocolHandler.

philippe44 committed:
fix small mp3 stream when using HTTPS and NB sockets

mherger committed:
Merge pull request #339 from philippe44/public/7.9

fix small mp3 stream when using HTTPS and NB sockets

mherger committed:
Changelog update

philippe44 committed:
enable non-blocking HTTPS on Windows

mherger committed:
Merge pull request #342 from philippe44/public/7.9

enable non-blocking HTTPS on Windows

philippe44 committed:
clear $! before check SSL connection (needed on Windows)

mherger committed:
Revert "enable non-blocking HTTPS on Windows"

This reverts commit bfda173ab73c40220e504a3f145e27fb859aeec8.

mherger committed:
Revert "Revert "enable non-blocking HTTPS on Windows""

This reverts commit 531151a00f57b6cf664134c06db995ed2c801054.

mherger committed:
Merge pull request #344 from philippe44/public/7.9

clear $! before check SSL connection (needed on Windows)

philippe44 committed:
if connection fails, remove ourselves from select

update

update

parenthesis

parenthesis

mherger committed:
Merge pull request #345 from philippe44/public/7.9

if connection fails, remove ourselves from select

mherger committed:
Increase plugin download timeout to 20s

mherger committed:
Avoid re-downloading plugin files if they've already been successfully downloaded.

philippe44 committed:
application/octet-stream handling

mherger committed:
Merge pull request #349 from philippe44/public/7.9

application/octet-stream handling

mherger committed:
Changelog update

mherger committed:
Remove left over debug statement.

mherger committed:
Merge branch 'public/7.7' into public/7.9

philippe44 committed:
small streams handling

- http_read_body shall return immediately when there is nothing to read instead of proceeding to avoid invoking functions & callbacks with empty content. This happens when using NB sockets
- streamAudioData must be protected when called with no content (in addition of the above) so that the "first" call happens with actual content. Otherwise expected parsing length was not set
- When stream is below 16kB, streamAudioData will never receive enough data, so parsing ID3v2 and bitrate are not done. But worse, when streamAudioData is supposed to launch actual HTTPS streaming connection, then streaming never starts

(cherry picked from commit 56f479010e12dff16ab0b297994030c89bcc117f)

philippe44 committed:
Support WAV as a stream format

philippe44 committed:
pcm (wav) is not seekable

I'm not sure sure exactly where is the right place to set that. I would have preferred to set a flag when the stream format is checked to avoid spreading the knowledge of format idiosyncrasies everywhere but I could not find a better idea

philippe44 committed:
forgot to set channel in header parsing (ogg & wav)

philippe44 committed:
add AIF streaming

To complete these patches, just adding AIF as a streamable format. AFAIK, it was just producing white noise before (on my Boom at least).
Only remaining part would be the possibility to strip out the header when streaming proxied sending pcm (not aif/wav). It's not clean to have it work by chance because header is a multiple of samplesize*channels but I really don't think there is a way to fix that as the audio streaming starts before the header check has been done, so we can't skip (or tell the player to skip in case of Direct) ... well this works like that and wav/aif streaming is a corner case anyway :-)

comment typo

mherger committed:
Changelog update

mherger committed:
Fix #387: filter by role if tags are given in titles query

mherger committed:
Fix #227 Update player's playlist name when the currently playing playlist is renamed.

mherger committed:
#277 More playlist management updates

* always update the client's playlist reference on `playlist save`
* remove playlist references from clients on `playlists delete`

mherger committed:
Add a not about macOS 11 (Big Sur) support to changelog

mherger committed:
Fix #400 - Don't hard-code treble/bass settings to Boom's values, but read from client model.

mherger committed:
Only warn once per URL when https fails

mherger committed:
Add a warning to server.log if an outdated IO::Socket::SSL library is found.

mherger committed:
v7.9.4

mherger committed:
Changelog update

mherger committed:
Fix FTS for tracks in playlists

As the `id` is no longer in the index we can't use `match` to find FTS records for a given track.

And no need to even iterate through tracks without a `file://` url. We can filter them out in the SQL query already.

mherger committed:
iTunes' "Downloaded" playlist can be huge - don't import it by default.

michaelherger commented:
It would be easier to compare the other way around. Your changes seem minimal and of only two types (of which likely only one is required for MySQL compatibility). Even if you had to start from scratch it would be quite straightforward to re-apply them. Certainly easier than reviewing a thousand changes files 😀.

mherger committed:
Update TIDAL NL localization

mherger committed:
Merge branch 'public/7.7' into public/7.9

# Conflicts:
#	Slim/Plugin/WiMP/strings.txt

mherger committed:
Get off of repos.squeezecommunity.org

- host server update info on downloads.slimdevices.com as long as we have the installers there anyway
- pull plugin repository `extensions.xml` directly from the LMS-Community repository

mherger committed:
Add a note about the updated SSL support for Windows

mherger committed:
Fix #440: Include icon element as image if there's no image
